### ISSUE TITLE
Android 7.0 SDK24 IllegalArgumentException

### ISSUE THREAD

**Issue by qxmwrr** [square/picasso#1494]
Why when the SDK is greater than 23 to call order upside 

```
private void setThreadCount(int threadCount) {
    if(VERSION.SDK_INT > 23) {
        this.setMaximumPoolSize(threadCount);
        this.setCorePoolSize(threadCount);
    } else {
        this.setCorePoolSize(threadCount);
        this.setMaximumPoolSize(threadCount);
    }

}

public void setMaximumPoolSize(int maximumPoolSize) {
    if (maximumPoolSize <= 0 || maximumPoolSize < corePoolSize)
        throw new IllegalArgumentException();
    this.maximumPoolSize = maximumPoolSize;
    if (workerCountOf(ctl.get()) > maximumPoolSize)
        interruptIdleWorkers();
}
```

```
E/AndroidRuntime: FATAL EXCEPTION: Picasso-DispatcherProcess: PID: 13274
java.lang.IllegalArgumentException
  at java.util.concurrent.ThreadPoolExecutor.setMaximumPoolSize(ThreadPoolExecutor.java:1659)
  at com.squareup.picasso.PicassoExecutorService.setThreadCount(PicassoExecutorService.java:83)
  at com.squareup.picasso.PicassoExecutorService.adjustThreadCount(PicassoExecutorService.java:45)
  at com.squareup.picasso.Dispatcher.performNetworkStateChange(Dispatcher.java:401)
  at com.squareup.picasso.Dispatcher$DispatcherHandler.handleMessage(Dispatcher.java:521)
  at android.os.Handler.dispatchMessage(Handler.java:102)
  at android.os.Looper.loop(Looper.java:154)
  at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by JakeWharton** [comment]
What version are you using?

**Comment by qxmwrr** [comment]
Sorry, I'm not sure, But it's may be 2.5.2.

**Comment by FabianTerhorst** [comment]
Im getting this with latest snapshot on nexus 6p with 7.0.

**Comment by davidtcdeveloper** [comment]
I am also having this issue here, in 2.6.0-SNAPSHOT.
Unfortunately, I cannot move down to 2.5.2 version because of the issues #1467 and #1450.
Do you know any workarround for this problem?

**Comment by qxmwrr** [comment]
resolve by commit 44e7f313d840de326b2cab4307a47efeac327fac

**Comment by ozv101** [comment]
@FabianTerhorst @davidtcdeveloper any fixes for this? are you still using 2.6.0-SNAPSHOT?
